addons/account/wizard/account_move_send.py

```diff
@@ -386,7 +411,12 @@ def _prepare_invoice_pdf_report(self, invoice, invoice_data):
         if invoice.invoice_pdf_report_id:
             return
 
-        content, _report_format = self.env['ir.actions.report']._render('account.account_invoices', invoice.ids)
+        # The default invoice pdf template (invoice with payments) is used unless another template is specified
+        pdf_template_name = DEFAULT_PDF_TEMPLATE_NAME
+        if 'pdf_template_id' in invoice_data:
+            pdf_template_name = invoice_data['pdf_template_id'].report_name
```

Comment:
Put `report_name` directly in invoice_data? Also, why not making sure to always have the `pdf_report_name` in `invoice_data` instead of checking everywhere `DEFAULT_PDF_TEMPLATE_NAME`?
In fact, since you added in `get_wizard_values`, I think `pdf_template_id` is always there.